Install fails, when Rprofile sets a working directory

I have a setup where `.Rprofile.site` has a `setwd` call in there, to have a fixed project folder from which to start. In this case I cannot seem to be able to install RcppParallel.

It seems like the configuration script tries load the configuration script here:

https://github.com/RcppCore/RcppParallel/blob/cbd61a1da2c564cd8463d101fe282b2f96a9ad09/tools/config.R#L582-L584

but because the working directory is set, `tools/config/configure.R` is not found, the code is not configured, and thus fails to install. 

I wonder if it would be more robust, if that script  could set its own working directory correctly when called, so external `setwd` values wouldn't interfere with the installation step.

The failure:

```
Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)
trying URL 'https://cloud.r-project.org/src/contrib/RcppParallel_5.0.1.tar.gz'
Content type 'application/x-gzip' length 1265516 bytes (1.2 MB)
==================================================
downloaded 1.2 MB

* installing *source* package ‘RcppParallel’ ...
** package ‘RcppParallel’ successfully unpacked and MD5 sums checked
** using staged installation
** preparing to configure package 'RcppParallel' ...
** finished configure for package 'RcppParallel'
** libs
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG     -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-V28x5H/r-base-3.6.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c init.cpp -o init.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG     -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-V28x5H/r-base-3.6.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c options.cpp -o options.o
options.cpp:2:10: fatal error: RcppParallel.h: No such file or directory
 #include <RcppParallel.h>
          ^~~~~~~~~~~~~~~~
compilation terminated.
/usr/lib/R/etc/Makeconf:177: recipe for target 'options.o' failed
make: *** [options.o] Error 1
ERROR: compilation failed for package ‘RcppParallel’
* removing ‘/usr/local/lib/R/site-library/RcppParallel’
Warning in install.packages :
  installation of package ‘RcppParallel’ had non-zero exit status
```

Notice how the configuration setup is too short. In the correct case it would be something like:

```
...
** preparing to configure package 'RcppParallel' ...
*** executing 'R CMD config'
*** executing 'R CMD config'
*** configured file: 'src/Makevars.in' => 'src/Makevars'
** finished configure for package 'RcppParallel'
...
```

@kevinushey to reproduce (I just reproduced simply)

1. Add a `setwd` into `~/.Rprofile` or `/etc/R/Rprofile.site`, that is:

```shell
# contents of ~/.Rprofile
setwd("/tmp")
```

2. try to install the package:

```
> getwd()
[1] "/tmp"
> install.packages('RcppParallel')
Installing package into '/usr/local/lib/R/site-library'
(as 'lib' is unspecified)
trying URL 'https://cloud.r-project.org/src/contrib/RcppParallel_5.0.1.tar.gz'
Content type 'application/x-gzip' length 1265516 bytes (1.2 MB)
==================================================
downloaded 1.2 MB

* installing *source* package 'RcppParallel' ...
** package 'RcppParallel' successfully unpacked and MD5 sums checked
** preparing to configure package 'RcppParallel' ...
** finished configure for package 'RcppParallel'
** libs
g++  -I/usr/share/R/include -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-AitvI6/r-base-3.4.4=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c init.cpp -o init.o
g++  -I/usr/share/R/include -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-AitvI6/r-base-3.4.4=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c options.cpp -o options.o
options.cpp:2:10: fatal error: RcppParallel.h: No such file or directory
 #include <RcppParallel.h>
          ^~~~~~~~~~~~~~~~
compilation terminated.
/usr/lib/R/etc/Makeconf:168: recipe for target 'options.o' failed
make: *** [options.o] Error 1
ERROR: compilation failed for package 'RcppParallel'
* removing '/usr/local/lib/R/site-library/RcppParallel'

The downloaded source packages are in
	'/tmp/RtmpeeXTJR/downloaded_packages'
Warning message:
In install.packages("RcppParallel") :
  installation of package 'RcppParallel' had non-zero exit status
```

I'm researching a fix, but so far I'm having some blockers on finding out within R scripts the folders where the scripts are running, which would be the key, I believe to the correction. That is, setting the working directory in `config.R` at the right stage, so the relative folders for the other invoked R script would be good. But this is so far my understanding. Any other opinion that is more expert on R than I am is very much welcome.

--- a/tools/config.R
+++ b/tools/config.R
@@ -574,6 +574,15 @@ if (!interactive()) {
     args <- commandArgs(TRUE)
     type <- args[[1]]
 
+    # preserve working directory
+    owd <- getwd()
+    on.exit(setwd(owd), add = TRUE)
+
+    # switch working directory to the calling scripts's directory as set
+    # by the shell, in case the R working directory was set to something else
+    basedir <- Sys.getenv("PWD", unset = ".")
+    setwd(basedir)
+
     # report start of execution
     package <- Sys.getenv("R_PACKAGE_NAME", unset = "<unknown>")
     fmt <- "** preparing to %s package '%s' ..."

